drivers/i2c/i2c_common.c

```diff
@@ -34,12 +34,19 @@ void i2c_dump_msgs_rw(const char *name, const struct i2c_msg *msgs,
 		const bool is_read = msg->flags & I2C_MSG_READ;
 		const bool dump_data = dump_read || !is_read;
 
-		LOG_DBG("   %c %s%s len=%02x: ", is_read ? 'R' : 'W',
-			msg->flags & I2C_MSG_RESTART ? "Sr " : "",
-			msg->flags & I2C_MSG_STOP ? "P" : "",
-			msg->len);
-		if (dump_data) {
-			LOG_HEXDUMP_DBG(msg->buf, msg->len, "contents:");
+		if (msg->len == 1 && dump_data) {
```

Comment:
Just curious: how would you feel about logging up to 4 bytes as a one-liner?

Reply:
I thought about that, but it ends up being a lot of code. Logging must be done with one log call, so you need to duplicate the entire format and args five times, for 1, 2, 3, 4, and (0 || >4 ) bytes.  It's an ugly heap of copy-pasta.

Putting one byte messages on the same line gives the easy 90% improvement, at least with the I2C devices I was using.  And it was enough to allow USB CDC ACM logging to keep up with I2C, which it couldn't do before.